lib/dpl/provider/pages.rb

```diff
@@ -45,9 +49,15 @@ def slug
       end
 
       def check_auth
+        unless @gh_token || @gh_deploy_key
+          error "must specify github-token or deploy-key"
+        end
       end
 
       def needs_key?
+        # Provider's generic support for keys is about provisioning and
```

Comment:
Is this documented somewhere? This should probably return `true` if there's no `gh-token` provided.

Reply:
This is still unresolved